c/secp256k1_blake160_multisig_all.c

```diff
@@ -0,0 +1,226 @@
+ *
+ */
+
+int main() {
+  int ret;
+  volatile uint64_t len = 0;
+  unsigned char temp[TEMP_SIZE];
+
+  /* Load args */
+  unsigned char script[SCRIPT_SIZE];
+  len = SCRIPT_SIZE;
+  ret = ckb_load_script(script, &len, 0);
+  if (ret != CKB_SUCCESS) {
+    return ERROR_SYSCALL;
+  }
+  if (len > SCRIPT_SIZE) {
+    return ERROR_SCRIPT_TOO_LONG;
+  }
+  mol_pos_t script_pos;
+  script_pos.ptr = (const uint8_t*)script;
+  script_pos.size = len;
+
+  mol_read_res_t args_res = mol_cut(&script_pos, MOL_Script_args());
+  if (args_res.code != 0) {
+    return ERROR_ENCODING;
+  }
+  mol_read_res_t args_bytes_res = mol_cut_bytes(&args_res.pos);
+  if (args_bytes_res.code != 0) {
+    return ERROR_ENCODING;
+  } else if (args_bytes_res.pos.size != BLAKE160_SIZE) {
+    return ERROR_ARGUMENTS_LEN;
+  }
+
+  /* Load witness of first input */
+  unsigned char witness[WITNESS_SIZE];
+  len = WITNESS_SIZE;
+  ret = ckb_load_witness(witness, &len, 0, 0, CKB_SOURCE_GROUP_INPUT);
+  if (ret != CKB_SUCCESS) {
+    return ERROR_SYSCALL;
+  }
+  if (len > WITNESS_SIZE) {
+    return ERROR_WITNESS_TOO_LONG;
+  }
+  if (len < FLAGS_SIZE) {
+    return ERROR_WITNESS_TOO_SHORT;
+  }
+
+  /* Get flags */
+  uint8_t pubkeys_cnt = witness[3];
+  uint8_t threshold = witness[2];
+  uint8_t require_first_n = witness[1];
+  if (pubkeys_cnt == 0) {
+    return ERROR_INVALID_PUBKEYS_CNT;
+  }
+  if (threshold > pubkeys_cnt) {
+    return ERROR_INVALID_THRESHOLD;
+  }
+  if (threshold == 0) {
+    threshold = pubkeys_cnt;
+  }
+  if (require_first_n > threshold) {
+    return ERROR_INVALID_REQUIRE_FIRST_N;
+  }
+  size_t multisig_script_len = FLAGS_SIZE + PUBKEY_SIZE * pubkeys_cnt;
+  size_t required_witness_len = multisig_script_len + SIGNATURE_SIZE * threshold;
+  if (len < required_witness_len) {
+    return ERROR_WITNESS_TOO_SHORT;
+  }
+  size_t extra_witness_len = len - required_witness_len;
+
+  /* Check multisig script hash */
+  blake2b_state blake2b_ctx;
+  blake2b_init(&blake2b_ctx, BLAKE2B_BLOCK_SIZE);
+  blake2b_update(&blake2b_ctx, witness, multisig_script_len);
+  blake2b_final(&blake2b_ctx, temp, BLAKE2B_BLOCK_SIZE);
+
+  if (memcmp(args_bytes_res.pos.ptr, temp, BLAKE160_SIZE) != 0) {
+    return ERROR_MULTSIG_SCRIPT_HASH;
+  }
+
+  /* Load tx hash */
+  unsigned char tx_hash[BLAKE2B_BLOCK_SIZE];
+  len = BLAKE2B_BLOCK_SIZE;
+  ret = ckb_load_tx_hash(tx_hash, &len, 0);
+  if (ret != CKB_SUCCESS) {
+    return ERROR_SYSCALL;
+  }
+
+  /* Prepare sign message */
+  unsigned char message[BLAKE2B_BLOCK_SIZE];
+  blake2b_init(&blake2b_ctx, BLAKE2B_BLOCK_SIZE);
+  blake2b_update(&blake2b_ctx, tx_hash, BLAKE2B_BLOCK_SIZE);
+
+  /* Load extra witness of first input and all witnesses of other inputs */
+  if (extra_witness_len > 0) {
+    blake2b_update(&blake2b_ctx, &witness[required_witness_len], extra_witness_len);
+  }
+  size_t i = 1;
+  while (1) {
+    len = WITNESS_SIZE;
+    ret = ckb_load_witness(temp, &len, 0, i, CKB_SOURCE_INPUT);
```

Comment:
Script should only cover the witness in the group, please use `CKB_SOURCE_GROUP_INPUT`.